pkg/ast/spl/spl.peg

```diff
@@ -505,6 +442,170 @@ Start <- SPACE? initialSearch:(InitialSearchBlock) filterBlocks:(FilterBlock)* q
 
     return q, nil
 }
+/ SPACE? PIPE CMD_GENTIMES SPACE genTimesOption:(GenTimesOptionList) queryAggBlocks:(QueryAggergatorBlock)* SPACE? EOF {
+    var q ast.QueryStruct
+    q.PipeCommands = &structs.QueryAggregators{
+        PipeCommandType: structs.GenerateEventType,
+        GenerateEvent: &structs.GenerateEvent{
+            GenTimes: genTimesOption.(*structs.GenTimes),
+        },
+    }
+
+    if queryAggBlocks != nil {
+        queryAggSlice := queryAggBlocks.([]any)
+
+        if len(queryAggSlice) > 0 {
+            curQueryAgg := q.PipeCommands
+            // Link the remaining chains.
+            for i := range queryAggSlice {
+                queryAgg := queryAggSlice[i].(*structs.QueryAggregators)
+                curQueryAgg.Next = queryAgg
+
+                // Go to the end of this chain.
+                for ; curQueryAgg.Next != nil; curQueryAgg = curQueryAgg.Next {}
+            }
+        }
+    }
+
+    return q, nil
+}
+
+PartialTimestamp <- [0-9][0-9] '/' [0-9][0-9] '/' [0-9][0-9][0-9][0-9] {
+    timestamp := string(c.text)
+    completeTimestamp := fmt.Sprintf("%s:00:00:00", timestamp)
+
+    return completeTimestamp, nil
+}
+
+IntegerAsTime <- intStr:IntegerAsString {
+    timeOffset, err := strconv.ParseInt(intStr.(string), 10, 64)
+    if err != nil {
+        return "", fmt.Errorf("Spl peg: Error while converting the integer: %v", err)
+    }
+    finalTime, err := utils.ApplyOffsetToTime(timeOffset, utils.TMDay, time.Now())
+    if err != nil {
+        return "", fmt.Errorf("Spl peg: Error while applying the offset: %v", err)
+    }
+    daySnap := fmt.Sprintf("%v", utils.TMDay)
+    finalTime, err = utils.ApplySnap(daySnap, finalTime)
+    if err != nil {
+        return "", fmt.Errorf("Spl peg: Error while applying the snap: %v", err)
+    }
+
+    return finalTime.UnixMilli(), nil
+}
+
+ConvertedTime <- timeStamp:(FullTimeStamp / PartialTimestamp) {
+    unixEpochInMs, err := utils.ConvertCustomDateTimeFormatToEpochMs(timeStamp.(string))
+    if err != nil {
+        return "", fmt.Errorf("Spl peg: Error while converting the timestamp: %v", err)
+    }
+    return unixEpochInMs, nil
+}
+
+
+GenTimestamp <- epochInMilli:(ConvertedTime / IntegerAsTime) {
+    return epochInMilli, nil
+}
+
+GenTimesOptionEnd <- "end" EQUAL timeStamp:(GenTimestamp) {
+    return &GenTimesOptionArgs{
+        argOption: "end",
+        genTimesOption: &structs.GenTimes{
+            EndTime: uint64(timeStamp.(int64)),
+        },
+    }, nil
+}
+
+GenTimesOptionStart <- "start" EQUAL timeStamp:(GenTimestamp) {
+    return &GenTimesOptionArgs{
+        argOption: "start",
+        genTimesOption: &structs.GenTimes{
+            StartTime: uint64(timeStamp.(int64)),
+        },
+    }, nil
+}
+
+GenTimesOptionIncrement <- "increment" EQUAL intStr:IntegerAsString unitStr:("s" / "m" / "d" / "h")? {
+    spanNum, err := strconv.ParseInt(intStr.(string), 10, 64)
```

Comment:
Is the `unitStr` Optional? when Increment is given? No right?

Reply:
It seems mandatory as per the Splunk documentation but upon testing on Splunk, it seems that if only a number is present second is used as the time unit. So, we are also making it optional similar to it.